designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -222,16 +224,21 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
   private def fetchGroupByGroup(
       basicProcesses: List[ScenarioWithDetails]
   )(implicit ec: ExecutionContext): FutureE[List[ScenarioWithDetails]] = {
-    basicProcesses
+    val groupedBasicProcesses = basicProcesses
       .map(_.name)
       .grouped(config.batchSize)
-      .foldLeft(EitherT.rightT[Future, NuDesignerError](List.empty[ScenarioWithDetails])) {
-        case (acc, processesGroup) =>
-          for {
-            current <- acc
-            fetched <- fetchProcessesDetails(processesGroup)
-          } yield current ::: fetched
-      }
+      .toVector
+    // We create ParVector manually instead of calling par for compatibility with Scala 2.12
+    val parallelCollection = new ParVector(groupedBasicProcesses)
+    val taskSupport        = new ExecutionContextTaskSupport(ec)
+    parallelCollection.tasksupport = taskSupport
```

Comment:
nit: you can make it in just one line, `parallelCollection.tasksupport = new ExecutionContextTaskSupport(ec)`
(or close this thread if you don't want to)

Reply:
Oh, I missed it! 
I inlined ExecutionContextTaskSupport instantiation.